Lib/test/test_urlparse.py

```diff
@@ -886,10 +866,43 @@ def test_parse_qsl_encoding(self):
     def test_parse_qsl_max_num_fields(self):
         with self.assertRaises(ValueError):
             urllib.parse.parse_qs('&'.join(['a=a']*11), max_num_fields=10)
-        with self.assertRaises(ValueError):
-            urllib.parse.parse_qs(';'.join(['a=a']*11), max_num_fields=10)
         urllib.parse.parse_qs('&'.join(['a=a']*10), max_num_fields=10)
 
+    def test_parse_qs_separator(self):
+        semicolon_cases = [
+            (";", {}),
+            (";;", {}),
+            (";a=b", {'a': ['b']}),
+            ("a=a+b;b=b+c", {'a': ['a b'], 'b': ['b c']}),
+            ("a=1;a=2", {'a': ['1', '2']}),
+            (b";", {}),
+            (b";;", {}),
+            (b";a=b", {b'a': [b'b']}),
+            (b"a=a+b;b=b+c", {b'a': [b'a b'], b'b': [b'b c']}),
+            (b"a=1;a=2", {b'a': [b'1', b'2']}),
+        ]
+        for orig, expect in semicolon_cases:
+            result = urllib.parse.parse_qs(orig, separator=';')
+            self.assertEqual(result, expect, "Error parsing %r" % orig)
+
+
+    def test_parse_qsl_separator(self):
+        semicolon_cases = [
+            (";", []),
+            (";;", []),
+            (";a=b", [('a', 'b')]),
+            ("a=a+b;b=b+c", [('a', 'a b'), ('b', 'b c')]),
+            ("a=1;a=2", [('a', '1'), ('a', '2')]),
+            (b";", []),
+            (b";;", []),
+            (b";a=b", [(b'a', b'b')]),
+            (b"a=a+b;b=b+c", [(b'a', b'a b'), (b'b', b'b c')]),
+            (b"a=1;a=2", [(b'a', b'1'), (b'a', b'2')]),
+        ]
+        for orig, expect in semicolon_cases:
+            result = urllib.parse.parse_qsl(orig, separator=';')
+            self.assertEqual(result, expect, "Error parsing %r" % orig)
+
```

Comment:
Perhaps we need more tests to ensure that the new behavior works (ignores semicolon). I wrote some test cases [here](https://github.com/python/cpython/pull/24271/files#diff-bb89d64e7413a637b421609f9a4f532c294037171c3f0150479930acf30425faR27)

Reply:
Definitely, will implement this. I still think we should co-author this PR, lmk how you want it to be done.

Reply:
I don't mind either way. The [co-author thing](https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors) is just for credit and you can place it in the [PR's first message](https://github.com/python/cpython/pull/24297#issue-559935639) . I honestly don't mind if you don't though :P .

Reply:
Added you to the main description. Also added the tests mentioned [here](https://github.com/python/cpython/pull/24297/files#diff-bb89d64e7413a637b421609f9a4f532c294037171c3f0150479930acf30425faR35).

Reply:
Ah oops, I was wondering why the co-author thing looked so weird, then I realised I forgot to give my email, sorry about that. The line should be `Co-authored-by: Ken Jin <28750310+Fidget-Spinner@users.noreply.github.com>`.

BTW, big thanks for all your efforts on this. I think this fix LGTM in general, I'm just waiting for Senthil's decision on the bug tracker :).

PS: You can mark this as resolved now, I don't see the button to do it on my end.